components/builder-api-client/src/builder.rs

```diff
@@ -793,10 +824,8 @@ impl BuilderAPIProvider for BuilderAPIClient {
                                     u.set_query(Some(&format!("target={}", target)))
                                 });
 
-        match self.download(req_builder, dst_path.as_ref(), token, progress) {
-            Ok(file) => Ok(PackageArchive::new(file)),
-            Err(e) => Err(e),
-        }
+        let file = self.download(req_builder, dst_path.as_ref(), token, progress)?;
+        Ok(PackageArchive::new(file))
```

Comment:
Re: error_from_status - as mentioned in a different comment, there are implications of compatibility and higher level changes, so I think it will be better in a future PR rather than undertaking this now